Gems/Atom/RHI/Vulkan/Code/Source/RHI/PipelineLibrary.cpp

```diff
@@ -108,5 +108,11 @@ namespace AZ
                 Debug::SetNameToObject(reinterpret_cast<uint64_t>(m_nativePipelineCache), name.data(), VK_OBJECT_TYPE_PIPELINE_CACHE, static_cast<Device&>(GetDevice()));
             }
         }
+
+        bool PipelineLibrary::SaveSerializedDataInternal([[maybe_unused]] const AZStd::string& filePath) const
+        {
+            //Vulkan drivers cannot save serialized data
+            return false;
```

Comment:
If I understand it correctly, m_isPsoCacheFileOperationsNeeded also determines if we will use this function.
We can make an assert for m_isPsoCacheFileOperationsNeeded, so that we don't accidentally fall into this if branch.

Reply:
Same for DX12.